test/asm-parser-subclass-integration-tests.ts

```diff
@@ -0,0 +1,189 @@
+                        throw error;
+                    }
+                }
+            }
+        });
+
+        it('should handle EWAVR assembly files from filter tests', () => {
+            try {
+                const ewavrAsmContent = readFileSync(resolve('test/filters-cases/ewarm-1.asm'), 'utf8');
+                const parser = new AsmEWAVRParser(properties.fakeProps({}));
+
+                const result = parser.processAsm(ewavrAsmContent, {
+                    directives: true,
+                    labels: true,
+                    commentOnly: false,
+                });
+
+                expect(result.asm.length).toBeGreaterThan(0);
+                // AsmEWAVRParser has custom processAsm that doesn't return labelDefinitions
+                expect(result.labelDefinitions).toBeUndefined();
+            } catch (error) {
+                // If file doesn't exist, skip this test
+                if ((error as any).code !== 'ENOENT') {
+                    throw error;
+                }
+            }
+        });
+    });
+
+    describe('Method override interactions', () => {
+        it('should use VcAsmParser labelFindFor override in findUsedLabels', () => {
+            const parser = new VcAsmParser();
+            const asmLines = ['_start:', 'mov eax, OFFSET _data', 'call _function', 'jmp _start'];
+
+            const usedLabels = parser.findUsedLabels(asmLines, true);
+
+            // Should find all labels using VC-specific regex
+            expect(usedLabels.has('_data')).toBe(true);
+            expect(usedLabels.has('_function')).toBe(true);
+            expect(usedLabels.has('_start')).toBe(true);
+        });
+
+        it('should use SPIRVAsmParser getUsedLabelsInLine override in processAsm', () => {
+            const parser = new SPIRVAsmParser();
+            const spirvCode = `
+                %main = OpFunction %void None %1
+                %entry = OpLabel
+                OpFunctionCall %void %helper_func
+                OpBranch %exit_label
+                %exit_label = OpLabel
+                OpReturn
+                OpFunctionEnd
+            `;
+
+            const result = parser.processAsm(spirvCode, {
+                directives: false,
+                labels: false,
+                commentOnly: false,
+            });
+
+            // Should detect SPIR-V specific labels using custom override
+            const hasSpirVLabels = result.asm.some(line => line.labels?.some(label => label.name === '%helper_func'));
+            expect(hasSpirVLabels).toBe(true);
+        });
+
+        it('should use AsmEWAVRParser labelFindFor override in processing', () => {
+            const parser = new AsmEWAVRParser(properties.fakeProps({}));
+            const asmLines = ['main:', 'ldi r16, HIGH(data_table)', 'call subroutine', 'rjmp main'];
+
+            const usedLabels = parser.findUsedLabels(asmLines, true);
+
+            // Current behavior: EWAVR labelFindFor() is broken - finds no labels
```

Comment:
EWAVR should _NOT_ be broken. We are likely not testing this correctly. We will work on this together.

Reply:
It turned out EWAVR _is_ broken: see #7781